app/controllers/purchases_controller.rb

```diff
@@ -106,8 +35,15 @@ def create
       render json: { errors: [{ message: 'Gateway not supported!' }] }, status: :unprocessable_entity
     end
   end
+end
+
+private
 
-  private
+  GATEWAYS = %w[paypal stripe]
+
+  def any_gateway?
```

Comment:
o nome desse método não corresponde com o que ele faz, tu não busca qualquer gateway mas sim os validos
```suggestion
  def valid_gateway?
```